[no-land] try fix

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#31927 [no-land] try fix**
* #30924 Reimplement c10::apply / c10::invoke with c++14

Differential Revision: [D19305405](https://our.internmc.facebook.com/intern/diff/D19305405/)

smessmer committed:
[no-land] try fix



Differential Revision: [D19305405](https://our.internmc.facebook.com/intern/diff/D19305405/)

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 84debae2:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=84debae25f659a84ebcb5bd095bde28d5596d242).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4163266) pytorch_windows_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=81572357) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=81572357))
<details>
<summary>
<code>FAILED: c10/test/CMakeFiles/c10_TypeTraits_test.dir/util/TypeTraits_test.cpp.obj</code>
</summary>

```
 
d_party\protobuf\src /DWIN32 /D_WINDOWS /GR  /w /EHa /MP /bigobj /MD /O2 /Ob2 /DNDEBUG /w /EHa /MP /bigobj   /MP /wd4018 /wd4065 /wd4146 /wd4244 /wd4251 /wd4267 /wd4305 /wd4307 /wd4309 /wd4334 /wd4355 /wd4506 /wd4800 /wd4996 /bigobj /showIncludes /Fothird_party\protobuf\cmake\CMakeFiles\libprotobuf-lite.dir\__\src\google\protobuf\arenastring.cc.obj /Fdthird_party\protobuf\cmake\CMakeFiles\libprotobuf-lite.dir\libprotobuf-lite.pdb /FS -c ..\third_party\protobuf\src\google\protobuf\arenastring.cc 
Microsoft (R) C/C++ Optimizing Compiler Version 19.11.25548.2 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

rty\googletest\googlemock\include -I..\third_party\googletest\googlemock -I..\third_party\googletest\googletest\include -I..\third_party\googletest\googletest /DWIN32 /D_WINDOWS /GR  /w /EHa /MP /bigobj -openmp /MD /O2 /Ob2  /w /EHa /MP /bigobj   -DCUDA_HAS_FP16=1 -DUSE_GCC_GET_CPUID -DUSE_AVX -DUSE_AVX2 -DTH_HAVE_THREAD -std:c++14 /showIncludes /Foc10\test\CMakeFiles\c10_typeid_test.dir\util\typeid_test.cpp.obj /Fdc10\test\CMakeFiles\c10_typeid_test.dir\ /FS -c ..\c10\test\util\typeid_test.cpp 
Microsoft (R) C/C++ Optimizing Compiler Version 19.11.25548.2 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

ooglemock\include -I..\third_party\googletest\googlemock -I..\third_party\googletest\googletest\include -I..\third_party\googletest\googletest /DWIN32 /D_WINDOWS /GR  /w /EHa /MP /bigobj -openmp /MD /O2 /Ob2  /w /EHa /MP /bigobj   -DCUDA_HAS_FP16=1 -DUSE_GCC_GET_CPUID -DUSE_AVX -DUSE_AVX2 -DTH_HAVE_THREAD -std:c++14 /showIncludes /Foc10\test\CMakeFiles\c10_TypeTraits_test.dir\util\TypeTraits_test.cpp.obj /Fdc10\test\CMakeFiles\c10_TypeTraits_test.dir\ /FS -c ..\c10\test\util\TypeTraits_test.cpp 
FAILED: c10/test/CMakeFiles/c10_TypeTraits_test.dir/util/TypeTraits_test.cpp.obj  
ooglemock\include -I..\third_party\googletest\googlemock -I..\third_party\googletest\googletest\include -I..\third_party\googletest\googletest /DWIN32 /D_WINDOWS /GR  /w /EHa /MP /bigobj -openmp /MD /O2 /Ob2  /w /EHa /MP /bigobj   -DCUDA_HAS_FP16=1 -DUSE_GCC_GET_CPUID -DUSE_AVX -DUSE_AVX2 -DTH_HAVE_THREAD -std:c++14 /showIncludes /Foc10\test\CMakeFiles\c10_TypeTraits_test.dir\util\TypeTraits_test.cpp.obj /Fdc10\test\CMakeFiles\c10_TypeTraits_test.dir\ /FS -c ..\c10\test\util\TypeTraits_test.cpp 
..\c10\test\util\TypeTraits_test.cpp(34): error C2338: 
..\c10\test\util\TypeTraits_test.cpp(35): error C2338: 
Microsoft (R) C/C++ Optimizing Compiler Version 19.11.25548.2 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

ocator.cpp.obj c10\cuda\CMakeFiles\c10_cuda.dir\impl\CUDAGuardImpl.cpp.obj c10\cuda\CMakeFiles\c10_cuda.dir\impl\CUDATest.cpp.obj  /out:bin\c10_cuda.dll /implib:LIB\c10_cuda.lib /pdb:bin\c10_cuda.pdb /dll /version:0.0 /machine:x64 /ignore:4049 /ignore:4217 /INCREMENTAL:NO  "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\lib\x64\cudart_static.lib"  LIB\c10.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cd ." 
Microsoft (R) Incremental Linker Version 14.11.25548.2 
Copyright (C) Microsoft Corporation.  All rights reserved. 
 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

smessmer committed:
Update on "[no-land] try fix"


Differential Revision: [D19305405](https://our.internmc.facebook.com/intern/diff/D19305405/)

[ghstack-poisoned]

smessmer committed:
Update on "[no-land] try fix"


Differential Revision: [D19305405](https://our.internmc.facebook.com/intern/diff/D19305405/)

[ghstack-poisoned]
